numba/tests/test_warnings.py

```diff
@@ -107,5 +106,18 @@ def do_loop(x):
             self.assertIn('lifted loops', str(w[2].message))
 
 
+    def test_deprecated(self):
+        @deprecated('foo')
+        def bar(): pass
+
+        with warnings.catch_warnings(record=True) as w:
```

Comment:
Think this block can be done in 1 line with an `assertRaisesRegex` and an `error` filter?